src/python/pants/backend/python/goals/run_pex_binary.py

```diff
@@ -37,7 +41,14 @@ async def create_pex_binary_run_request(
         ),
         Get(TransitiveTargets, TransitiveTargetsRequest([field_set.address])),
     )
-    transitive_targets = await Get(TransitiveTargets, TransitiveTargetsRequest([field_set.address]))
+
+    if not entry_point.val:
+        raise MissingEntryPoint(
+            "To run a `pex_binary`, you must either set the `entry_point` field or the `sources` "
+            f"field, but {field_set.address} left both fields off. See "
+            "https://www.pantsbuild.org/docs/python-package-goal#creating-a-pex-file-from-a-pex_binary-target "
+            "for how to set the entry point."
+        )
```

Comment:
It would be nice if the behavior of `./pants run` and `./pants package && ./run.pex` stayed relatively close to one another rather than diverging in terms of their requirements like this... would it involve duplication to use PEX's entrypoint behavior here too?

Reply:
Maybe it doesn't make sense to ban `run`. I forgot that we have passthrough args. You may want to be able to do `./pants run path/to:pex_binary -- -c 'some python code"` for example.